operatortrace-go/pkg/client/conditions.go

```diff
@@ -0,0 +1,224 @@
+		return err
+	}
+
+	newCondition := map[string]interface{}{
+		"Type":               conditionType,
+		"Status":             metav1.ConditionUnknown,
+		"LastTransitionTime": metav1.Now(),
+		"Message":            message,
+	}
+	conditions = append(conditions, newCondition)
+
+	return setConditionsFromMap(obj, conditions, scheme)
+}
+
+func deleteConditionAsMap(conditionType string, obj client.Object, scheme *runtime.Scheme) error {
+	// Retrieve the current conditions as a map
+	conditions, err := getConditionsAsMap(obj, scheme)
+	if err != nil {
+		return err
+	}
+
+	var outConditions []map[string]interface{}
+	for _, condition := range conditions {
+		// Check if "Type" key exists
+		conType, exists := condition["Type"]
+		if !exists {
+			return fmt.Errorf("condition does not contain a 'Type' field")
+		}
+
+		// Convert conType to string using reflection
+		conTypeStr, err := convertToString(conType)
+		if err != nil {
+			return fmt.Errorf("failed to convert 'Type' field to string: %v", err)
+		}
+
+		if conTypeStr != conditionType {
+			outConditions = append(outConditions, condition)
+		}
+	}
+
+	// Set the updated conditions back to the object
+	return setConditionsFromMap(obj, outConditions, scheme)
+}
+
+func getConditionsAsMap(obj client.Object, scheme *runtime.Scheme) ([]map[string]interface{}, error) {
+	gvk, err := apiutil.GVKForObject(obj, scheme)
+	if err != nil {
+		return nil, fmt.Errorf("problem getting the GVK: %w", err)
+	}
+
+	objTyped, err := scheme.New(gvk)
+	if err != nil {
+		return nil, fmt.Errorf("problem creating new object of kind %s: %w", gvk.Kind, err)
+	}
+
+	if err := scheme.Convert(obj, objTyped, nil); err != nil {
+		return nil, fmt.Errorf("problem converting object to kind %s: %w", gvk.Kind, err)
+	}
+
+	val := reflect.ValueOf(objTyped)
+	statusField := val.Elem().FieldByName("Status")
+	if !statusField.IsValid() {
+		return nil, fmt.Errorf("status field not found in kind %s", gvk.Kind)
+	}
+
+	conditionsField := statusField.FieldByName("Conditions")
+	if !conditionsField.IsValid() {
+		return nil, fmt.Errorf("conditions field not found in kind %s", gvk.Kind)
+	}
+
+	conditionsValue := conditionsField.Interface()
+	val = reflect.ValueOf(conditionsValue)
+	if val.Kind() != reflect.Slice {
+		return nil, fmt.Errorf("conditions field is not a slice")
+	}
+
+	var conditionsAsMap []map[string]interface{}
+	for i := 0; i < val.Len(); i++ {
+		conditionVal := val.Index(i)
+		if conditionVal.Kind() == reflect.Ptr {
+			conditionVal = conditionVal.Elem()
+		}
+
+		conditionMap := make(map[string]interface{})
+		for _, field := range reflect.VisibleFields(conditionVal.Type()) {
+			fieldValue := conditionVal.FieldByIndex(field.Index)
+			conditionMap[field.Name] = fieldValue.Interface()
+		}
+
+		conditionsAsMap = append(conditionsAsMap, conditionMap)
+	}
+
+	return conditionsAsMap, nil
+}
+
+func setConditionsFromMap(obj client.Object, conditionsAsMap []map[string]interface{}, scheme *runtime.Scheme) error {
```

Comment:
conditions has it's own schema do we have to convert it to a map instead of marshaling it after the kind is found to the status field if they have conditions?